src/main/java/mate/academy/bookshop/controller/AuthenticationController.java

```diff
@@ -0,0 +1,39 @@
+package mate.academy.bookshop.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookshop.dto.user.UserLoginRequestDto;
+import mate.academy.bookshop.dto.user.UserLoginResponseDto;
+import mate.academy.bookshop.dto.user.UserRegistrationRequestDto;
+import mate.academy.bookshop.dto.user.UserResponseDto;
+import mate.academy.bookshop.exception.RegistrationException;
+import mate.academy.bookshop.security.AuthenticationService;
+import mate.academy.bookshop.service.UserService;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Authentication management", description = "Endpoints for managing authentication")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/auth")
+public class AuthenticationController {
+    private final UserService userService;
+    private final AuthenticationService authenticationService;
+
+    @Operation(summary = "Login", description = "To login users")
+    @PostMapping("/login")
+    public UserLoginResponseDto login(@RequestBody UserLoginRequestDto requestDto) {
+        return authenticationService.authenticate(requestDto);
+    }
+
+    @Operation(summary = "Register", description = "To register users")
+    @PostMapping("/register")
+    public UserResponseDto register(@RequestBody @Valid UserRegistrationRequestDto request)
+            throws RegistrationException {
```

Comment:
remove throws

Reply:
I can't remove throws , because this code ensures that the register method either throws a RegistrationException or propagates it in the method signature as required by the method contract.